Support non-KV and KV v1/v2 secret engines

tpounds opened the pull request:
##### SUMMARY
Similar to the change in #41132 differs by querying the mount configuration first to select the appropriate hvac client methods. It also negates the need to insert an additional `/data` component in the lookup path.

Fixes #41132
Fixes #50598
Fixes #54462
Fixes #64150

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
hashi_vault lookup plugin

tpounds committed:
Support generic and KV v1/v2 secret engines.

Fixes #41132
Fixes #50598
Fixes #54462
Fixes #64150

tpounds committed:
Update KV v2 compatibility notes/usage docs.

tpounds committed:
Add tests for KV v1/v2 and generic secrets engines.

felixfontein commented:
You need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to) for this. Also, in case this changes the behavior of existing roles/playbooks using this for v2 secret engines, you need to document that in the [porting guide](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.10.rst#noteworthy-module-changes).

felixfontein commented:
Please note that integration tests for the lookup are failing: https://app.shippable.com/github/ansible/ansible/runs/149074/75/tests

tpounds committed:
Add changelog fragment.

tpounds committed:
Add KV v2 behavior changes to porting guide.

tpounds commented:
@felixfontein I've fixed the broken tests and added the relevant changelog fragment/porting guidelines. I've updated the behavior to be consistent with pr-#41132 to keeps the previous KV v2 secret path (e.g. `secret/data/:path`) but always return the latest version.

tpounds committed:
Fix pylint blacklisted name warnings.

felixfontein commented:
CC @Im0

samccann reviewed (approved):
Docs portion lgtm

Im0 commented:
Tested in `Vault v1.2.3` and worked for me with both kv1 and kv2 secrets.

Big thanks to @beudbeud and @tpounds .

@felixfontein LGTM

felixfontein commented:
Thanks everyone for working on this!
